themes/bootstrap3/templates/search/advanced/build_page.phtml

```diff
@@ -4,10 +4,12 @@ $(document).ready(function() {
   $('#groupPlaceHolder').removeClass('hidden');
   $('#new_search_template').addClass('hidden').detach().appendTo('[role="main"]');
   $('#new_search_template').find('.adv-search').removeAttr('id');
+  $('#new_search_template').find('.form-control').removeAttr('id');
```

Comment:
What exactly is going on here? I understand that we have an HTML template set up that we're copying to create the groups on the advanced search screen, and thus we need to remove the IDs in order to avoid HTML validation problems caused by duplicate IDs... and at least some of the IDs are there in the first place to allow labels to be associated with appropriate controls. But if we delete the IDs, aren't we breaking the labels? Has this whole thing been thought through all the way? @crhallberg, have you already looked into this? Do we need a better approach here to achieve labeling without risking duplicate IDs?

Reply:
The JS that copies the template onto the new search box also sets the appropriate IDs everywhere they were removed.

I think vufind/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AdvancedSearchTest.php at line 127 would fail if removing this ID was a regression.

Reply:
I saw this and saw no problems with it. Maybe we should remove the ID from the HTML to make this a little simpler. https://github.com/vufind-org/vufind/blob/23c75f5ef9477b3f921e9dbc58d07602018123c7/themes/bootstrap3/templates/search/advanced/layout.phtml#L108

Reply:
Reviewing the template brings back some memory of this. We shouldn't remove the id because this is serving two purposes -- it's not only the template html but also the non-Javascript fallback logic. That's probably something we should consider changing -- it's overcomplicated, for one thing -- but as it stands, if we took the id out without changing anything else, it would result in invalid labels for non-JS users.